File: framework/src/main/java/com/djangoogle/framework/widget/DjangoLoading.java
package com.djangoogle.framework.widget;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.djangoogle.framework.R;

/**
 * 基础等待层弹窗
 * Created by Djangoogle on 2018/10/24 09:33 with Android Studio.
 */
public class DjangoLoading extends DialogFragment {

	@Override
	public void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setStyle(DialogFragment.STYLE_NO_TITLE, R.style.django_loading);
	}

	@Nullable
	@Override
	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		return inflater.inflate(R.layout.dialog_loading_base, container, false);
	}

	@NonNull
	@Override
	public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
		Dialog dialog = super.onCreateDialog(savedInstanceState);
		dialog.setCancelable(false);
		dialog.setCanceledOnTouchOutside(false);
		return dialog;
	}
}

File: framework/src/main/kotlin/com/djangoogle/framework/widget/DjangoLoading.kt
package com.djangoogle.framework.widget

import android.app.Dialog
import android.os.Bundle
import android.view.KeyEvent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.djangoogle.framework.R
import com.djangoogle.framework.fragment.DjangoDialogFragment

/**
 * 基础等待层弹窗
 * Created by Djangoogle on 2018/10/24 09:33 with Android Studio.
 */
class DjangoLoading : DjangoDialogFragment() {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setStyle(STYLE_NO_TITLE, R.style.django_loading)
	}

	override fun onCreateView(
			inflater: LayoutInflater,
			container: ViewGroup?,
			savedInstanceState: Bundle?
	): View? {
		return inflater.inflate(R.layout.dialog_loading_django, container, false)
	}

	override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
		val djangoLoading = super.onCreateDialog(savedInstanceState)
		djangoLoading.setCancelable(false)
		djangoLoading.setCanceledOnTouchOutside(false)
		djangoLoading.setOnKeyListener { _, keyCode, _ -> KeyEvent.KEYCODE_BACK == keyCode }
		return djangoLoading
	}
}
